### ISSUE TITLE
CoreText garbles ligatures

### ISSUE THREAD

**Issue by denschub** [macvim-dev/macvim#520]
Snapshot 134, `macligatures` is enabled. The font I am using is PragmataPro. It's hard to describe what I am experiencing, so let's have a look at two gifs.

Here is entering `!==` without CoreText:

![not_coretext](https://user-images.githubusercontent.com/344777/27882373-b9a4a922-61cc-11e7-8bb2-789045a799d4.gif)

As you can see, everything looks fine. Here is the same with CoreText enabled:

![coretext](https://user-images.githubusercontent.com/344777/27882392-c86ce280-61cc-11e7-9fb3-d2c4ea095c34.gif)

As you can see, something is quite off here. The ligature gets broken somehow during working on the line. However, when changing the line, everything looks normal. Obviously, disabling CoreText is a perfect workaround, but since scrolling performance is somewhat horrible, I'd love to know if there is anything anyone can do here to resolve this rendering weirdness.

**Comment by Hovushka** [comment]
Yeah, it works fine in the terminal, but the GUI version (which is my favourite) still lacks proper support

**Comment by eirnym** [comment]
@Hovushka thank you for a reply. If you have any solution for this, you can try
